Include/internal/pycore_interp.h

```diff
@@ -41,6 +41,22 @@ struct _Py_long_state {
     int max_str_digits;
 };
 
+#define DICT_MAX_WATCHERS 8
+
+struct _Py_dict_interp_state {
```

Comment:
@colesbury 
More proper position might be pycore_dict.h but there are some header dependency issue.
So let's declare the _Py_dict_interp_state at here.

Reply:
Yeah, `pycore_dict.h` (or `pycore_dict_state.h`) seem like a better locations, but if there are technical reasons that is difficult, this does not seem like such a big deal either.

Reply:
The only real change is pulling the freelist out.  Why can't the remainder stay in pycore_dict_state,h?

Reply:
It's still possible, I guess :) I will revert about it.

Reply:
Thanks!